packages/dd-trace/src/exporters/common/docker.js

```diff
@@ -2,34 +2,64 @@
 
 const fs = require('fs')
 
+const { DD_EXTERNAL_ENV } = process.env
+
 // The second part is the PCF / Garden regexp. We currently assume no suffix($) to avoid matching pod UIDs
 // See https://github.com/DataDog/datadog-agent/blob/7.40.x/pkg/util/cgroups/reader.go#L50
 const uuidSource =
 '[0-9a-f]{8}[-_][0-9a-f]{4}[-_][0-9a-f]{4}[-_][0-9a-f]{4}[-_][0-9a-f]{12}|[0-9a-f]{8}(?:-[0-9a-f]{4}){4}$'
 const containerSource = '[0-9a-f]{64}'
 const taskSource = '[0-9a-f]{32}-\\d+'
+const lineReg = /^(\d+):([^:]*):(.+)$/
 const entityReg = new RegExp(`.*(${uuidSource}|${containerSource}|${taskSource})(?:\\.scope)?$`, 'm')
 
+const cgroup = readControlGroup()
 const entityId = getEntityId()
+const inode = getInode()
 
 function getEntityId () {
-  const cgroup = readControlGroup() || ''
-  const match = cgroup.trim().match(entityReg) || []
+  const match = cgroup.match(entityReg) || []
 
   return match[1]
 }
 
+function getInode () {
+  const match = cgroup.match(lineReg) || []
+
+  return readInode(match[3])
+}
+
 function readControlGroup () {
   try {
-    return fs.readFileSync('/proc/self/cgroup').toString()
+    return fs.readFileSync('/proc/self/cgroup').toString().trim()
+  } catch (err) {
+    return ''
+  }
+}
+
+function readInode (path) {
+  if (!path) return 0
+
+  const strippedPath = path.replace(/^\//, '').replace(/\/$/, '')
+
+  try {
+    return fs.statSync(`/sys/fs/cgroup/${strippedPath}`).ino
   } catch (err) {
-    // ignore
+    return 0
   }
 }
 
 module.exports = {
-  // can be the container ID but not always depending on the orchestrator
-  id () {
-    return entityId
+  inject (carrier) {
+    if (entityId) {
+      carrier['Datadog-Container-Id'] = entityId
```

Comment:
Is the `Id` case inconsistency intentional?

Reply:
@tlhunter Do you want me to make the change or it's fine as it is? And if I should make the change, what casing would you like to see?

Reply:
I suppose it's fine as is, especially if Python is doing it this way and with the agent being robust with its parsing.